code/modules/admin/verbs/map_export.dm

```diff
@@ -0,0 +1,261 @@
+/client/proc/map_export()
+	set category = "Debug"
+	set name = "Export Map"
+	set desc = "Select a part of the map by coordinates and download it."
+	
+	if(!check_rights(R_DEBUG))
+		return
+
+	var/z_level = input("Export Which Z-Level?", "Map Exporter", 2) as num
+	var/start_x = input("Start X?", "Map Exporter", 1) as num
+	var/start_y = input("Start Y?", "Map Exporter", 1) as num
+	var/end_x = input("End X?", "Map Exporter", world.maxx-1) as num
+	var/end_y = input("End Y?", "Map Exporter", world.maxy-1) as num
+	var/date = time2text(world.timeofday, "YYYY-MM-DD_hh-mm-ss")
+	var/file_name = sanitize_filename(input("Filename?", "Map Exporter", "exportedmap_[date]") as text)
+	var/confirm = tgui_alert(usr, "Are you sure you want to do this? This will cause extreme lag!", "Map Exporter", list("Yes", "No"))
+
+	if(confirm != "Yes")
+		return
+
+	var map_text = write_map(start_x, start_y, z_level, end_x, end_y, z_level)
+	text2file(map_text, "data/[file_name].dmm")
+	usr << ftp("data/[file_name].dmm", "[file_name].dmm")
+
+/proc/sanitize_filename(t)
+	return hashtag_newlines_and_tabs(t, list("\n"="", "\t"="", "/"="", "\\"="", "?"="", "%"="", "*"="", ":"="", "|"="", "\""="", "<"="", ">"=""))
+
+/proc/hashtag_newlines_and_tabs(text, list/repl_chars = list("\n"="#","\t"="#"))
+	for(var/char in repl_chars)
+		var/index = findtext(text, char)
+		while(index)
+			text = copytext(text, 1, index) + repl_chars[char] + copytext(text, index + length(char))
+			index = findtext(text, char, index + length(char))
+	return text
+
+/obj/proc/on_object_saved(var/depth = 0)
+	return ""
+
+// Save resources in silo
+/obj/machinery/ore_silo/on_object_saved(var/depth = 0)
+	if(depth >= 10)
+		return ""
+	var/data
+	var/datum/component/material_container/material_holder = GetComponent(/datum/component/material_container)
+	for(var/each in material_holder.materials)
+		var/amount = material_holder.materials[each] / 100
+		var/datum/material/material_datum = each
+		while(amount > 0)
+			var/amount_in_stack = max(1, min(50, amount))
+			amount -= amount_in_stack
+			data += "[data ? ",\n" : ""][material_datum.sheet_type]{\n\tamount = [amount_in_stack]\n\t}"
+	return data
+
+/**Map exporter
+* Inputting a list of turfs into convert_map_to_tgm() will output a string
+* with the turfs and their objects / areas on said turf into the TGM mapping format
+* for .dmm files. This file can then be opened in the map editor or imported
+* back into the game.
+* ============================
+* This has been made semi-modular so you should be able to use these functions
+* elsewhere in code if you ever need to get a file in the .dmm format
+**/
+/atom/proc/get_save_vars()
+	return list("pixel_x",
+				"pixel_y",
+				"dir",
+				"name",
+				"req_access",
+				"piping_layer", 
+				"color", 
+				"icon", 
+				"icon_state", 
+				"pipe_color", 
+				"amount", 
+				"dwidth",
+				"dheight", 
+				"height", 
+				"width", 
+				"roundstart_template", 
+				"shuttle_id"
+			)
```

Comment:
you need to properly multiline lists like these. some of these are fine in this PR but stuff like this should be done properly, like so.

```suggestion
	return list(
		"pixel_x",
		"pixel_y",
		"dir",
		"name",
		"req_access",
		"piping_layer", 
		"color", 
		"icon", 
		"icon_state", 
		"pipe_color", 
		"amount", 
		"dwidth",
		"dheight", 
		"height", 
		"width", 
		"roundstart_template", 
		"shuttle_id",
	)
```

This list should also be alphabetized.

Also, since this list can drift from variables that actually exist on `/atom`, please use the `NAMEOF()` helper macros to ensure that these variables get removed if a rework were to occur and something like `pipe_color` gets renamed to something else (it will cause a compile failure to compel the person to fix that mistake). You can read more on how to use `NAMEOF()` in it's documentation or in spots like `vv_edit_var()` proc chains.

Reply:
I'm not quite sure if I'm making the right decision.